### PR TITLE
Customer api

### PR DESCRIPTION
**Pull request by lemon037**
Created the new db script 15-10-2020 one, and all the stuff mentioned in the latest commit comment.

### CODE REVIEW

**Comment by hpy9707** [comment]
Nice work in redeign the modelling. I am shocked you can build a simple ORM in these days and it looks like very formal. 
From my point of view, the only thing I suggest is create a new Exception folder and put two exceptions from controller folder and resource folder

**Review by hpy9707 — COMMENTED** [review]
Nice work in redeign the model and make a simple ORM. Although the ORM is quite simple, but contains all key functions. 
My suggestion is just making a new folder called"Exception" and put two exception.py from Controller and Resource in this folder

File: app/Controller/CustomerController.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        cs.create_customer(new_customer, new_address)
+        return {'customer': new_customer.__dict__, 'address': new_address.__dict__}, 201
+
+    # Get customers list
+    elif request.method == 'GET':
+        all_customers = cs.list_all_customers()
+        return jsonify([c.__dict__ for c in all_customers]), 200
+
+
+@cust.route('/api/customer/<customer_id>', methods=['GET', 'DELETE'])
```
Comment (ansabkhaliq) [inline app/Controller/CustomerController.py:91]: do you know you don't have to define Get and Delete in 1 method? 
You can create another one just like

@cust.route('/api/customer/<customer_id>', Methods=[Get] 
If u call it with Get from frontend.. it come come to this one.. 
If you call it with delete from frontend, it won't call this one. 

I would suggest breaking it up. just copy paste and add Method = ['Get'] to one and Method = ['Delete'] to other...

Its not good practice to have If, else check in the controller to see what type of request it is..

File: app/Controller/CustomerController.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    if cust_code not in cs.get_used_customer_codes():
+        return {'message': f'Provided customer code {cust_code} does not match any customer'}, 404
+
+    # TODO this is a problematic function, to be fixed
+    sync_products(customer_code=cust_code)
+    return {'message': 'Switch customer successfully'}, 200
+
+
+@cust.route('/api/customers', methods=['GET', 'POST'])
```
Comment (ansabkhaliq) [inline app/Controller/CustomerController.py:45]: I left a comment below, would be great if you can fix this one as welll

**Review by ansabkhaliq — CHANGES_REQUESTED** [review]
There are some comments left by Andy and me. Shouldn't take you long. Can you please fix them? I will merge it after that.

File: app/Controller/CustomerController.py (hunk 1 above)
Comment (lemon037) [reply app/Controller/CustomerController.py:91]: Ok I'll make this change, I've considered that before, this could make the code looks cleaner.

Comment (ansabkhaliq) [reply app/Controller/CustomerController.py:91]: yup and professional as well.

**Review by ansabkhaliq — APPROVED** [review]

**Pull request merged**